notecard/notecard.go

```diff
@@ -46,14 +45,16 @@ type Context struct {
 	PortEnumFn	   func() (ports []string, err error)
 	PortDefaultsFn func() (port string, portConfig int)
 	CloseFn		   func(context *Context)
+	ReopenFn	   func(context *Context) (err error)
 	ResetFn		   func(context *Context) (err error)
 	TransactionFn  func(context *Context, reqJSON []byte) (rspJSON []byte, err error)
 
 	// Serial instance state
 	isSerial	   bool
-	openSerialPort *serial.Port // tarm
-	//openSerialPort io.ReadWriteCloser // jacobsa
-	PortName       string
+	openSerialPort *serial.Port
+	serialConfig 	serial.Config
+	i2cName		string
+	i2cAddress	int
 }
```

Comment:
This context used to be kept at a higher level, so we'd have to dig it up again if we wanted to close and reopen a serial port.